### DISCUSSION TITLE
Server for HFS - Windows vs Linux

### DISCUSSION THREAD

**Discussion by Officepoint** [rejetto/hfs#461] [General]
Hi Rejetto!

I would like to ask you what environment as a host do you use for HFS,
Windows or Linux ( Server, Ubuntu, Mint, etc)?
I have prepared a laptop for HFS with 2 ssd hdd of 512 gb
each, having W11 latest version. I know from the experience of more
advanced users than me, that W10-W11 is not very good for server, and
Linux is much better in this direction at least for security. I would
look for a version of Linux to install but I think I will surely run
into other OS specific problems. I am more familiar with Windows than
Linux.
However, regardless of the hosting solution, a backup solution, if it
is not automatic, at least manual, is imperative.
I assume that all settings made in the Admin Panel are saved in the
HFS folder and to have a full backup it is sufficient to copy the
whole folder, correct?
What do you think, do you have any suggestions on hosting for HFS?

Thanks!

**Comment by rejetto** [comment]
I guess it depends if you need a small personal server or not. If you are small, I don't see anything wrong with using Windows, but i'm no expert on this.
Linux better as a potential, but one of the pieces of the system is you who manage and configure it, and if you don't know or learn enough of linux, the final result may not be "much better" but even worse :)

i have a laptop with windows 11. Recently i've been forced to shut it down for battery problems. I don't exclude replacing it with a raspberry pi.

When hfs starts, it says "cwd", and that's where it saves stuff.cwd is short for current working directory.hfs.exe uses its same folder, because that's what people is used to, but all other systems save to a different folder, because that's what's normal in POSIX systems.
Maybe cwd is too technical and I should just say "Working directory" in the output.

**Comment by Officepoint** [comment]
Thanks, these are valid points!
All the best!
